portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,89 +20,116 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
 import com.google.gson.Gson;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime; 
+import java.lang.Math;
 
-/** Servlet that handles comment data. */
+/** 
+  * Servlet that uploads and retrieves persistent comment data using datastore.
+  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Represents a single comment */
+  private static class Comment {
```

Comment:
FYI - only as a stretch goal, not required for this project. 

When considering where to place logic, serialization and deserialization is often a task that needs done repeatedly. This tells us that it's likely a good idea to put somewhere easily accessible. 

Adding a toEntity() and a fromEntity(Entity e) method could improve readability. I'll add an inline comment below with rationale.

Reply:
I actually really like this idea! I've added toEntity and fromEntity; it makes much more sense this way :)